### PR TITLE
Add routing table tests in eos_validate_state role (lo0 and lo1 addresses check)

### LINKED ISSUE #560: add a test in eos_validate_state role to validate lo0 and lo1 addresses are in the routing table
**Issue Type**

- Role enhancement

add a test in eos_validate_state role to validate lo0  and lo1 addresses are in the routing table

working
```
DC1-LEAF1A#sh ip route 192.168.254.5 | json
{
    "vrfs": {
        "default": {
            "routes": {
                "192.168.254.5/32": {
                    "kernelProgrammed": true,
                    "directlyConnected": false,
                    "routeAction": "forward",
                    "routeLeaked": false,
                    "vias": [
                        {
                            "interface": "Ethernet1",
                            "nexthopAddr": "172.31.255.0"
                        },
                        {
                            "interface": "Ethernet2",
                            "nexthopAddr": "172.31.255.2"
                        }
                    ],
                    "metric": 110,
                    "hardwareProgrammed": true,
                    "routeType": "ISISLevel2",
                    "preference": 115
                }
            },
            "allRoutesProgrammedKernel": true,
            "routingDisabled": false,
            "allRoutesProgrammedHardware": true,
            "defaultRouteState": "notSet"
        }
    }
}
DC1-LEAF1A#sh ip route 192.168.254.5

VRF: default
Codes: C - connected, S - static, K - kernel, 
       O - OSPF, IA - OSPF inter area, E1 - OSPF external type 1,
       E2 - OSPF external type 2, N1 - OSPF NSSA external type 1,
       N2 - OSPF NSSA external type2, B - BGP, B I - iBGP, B E - eBGP,
       R - RIP, I L1 - IS-IS level 1, I L2 - IS-IS level 2,
       O3 - OSPFv3, A B - BGP Aggregate, A O - OSPF Summary,
       NG - Nexthop Group Static Route, V - VXLAN Control Service,
       DH - DHCP client installed default route, M - Martian,
       DP - Dynamic Policy Route, L - VRF Leaked,
       RC - Route Cache Route

 I L2     192.168.254.5/32 [115/110] via 172.31.255.0, Ethernet1
                                     via 172.31.255.2, Ethernet2

DC1-LEAF1A#
```
not working
```
DC1-LEAF1A#sh ip route 192.168.254.1

VRF: default
Codes: C - connected, S - static, K - kernel, 
       O - OSPF, IA - OSPF inter area, E1 - OSPF external type 1,
       E2 - OSPF external type 2, N1 - OSPF NSSA external type 1,
       N2 - OSPF NSSA external type2, B - BGP, B I - iBGP, B E - eBGP,
       R - RIP, I L1 - IS-IS level 1, I L2 - IS-IS level 2,
       O3 - OSPFv3, A B - BGP Aggregate, A O - OSPF Summary,
       NG - Nexthop Group Static Route, V - VXLAN Control Service,
       DH - DHCP client installed default route, M - Martian,
       DP - Dynamic Policy Route, L - VRF Leaked,
       RC - Route Cache Route

Gateway of last resort is not set


DC1-LEAF1A#sh ip route 192.168.254.1 | json 
{
    "vrfs": {
        "default": {
            "routes": {},
            "allRoutesProgrammedKernel": true,
            "routingDisabled": false,
            "allRoutesProgrammedHardware": true,
            "defaultRouteState": "notSet"
        }
    }
}
DC1-LEAF1A#
```

### PR DESCRIPTION
**Pull request by ksator**
## Change Summary

Add routing table tests in eos_validate_state role (to check if lo0 and lo1 addresses are in the routing table). 
This is based on devices type.  
This task can be extended to test others IP destination.

 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

 fixes #560 
## Component(s) name

update roles `eos_validate_state`
- create task routing_table.yml  
- update task main.yml  
- update readme 
- update template generate_vars_for_testing.j2 
- update template validate_state_report-csv.j2 

## How to test

run the role `eos_validate_state` with a working fabric. 
break somnething (routing on a device) and re-run the role `eos_validate_state` (you will see errors reported)   
 
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2 [hunk 1]
```diff
@@ -8,4 +8,16 @@
 {%     endif %}
 {% endfor %}
 loopback0_reachability:
-  loopback0_range: {{ address | unique }}
\ No newline at end of file
+  loopback0_range: {{ address | unique }}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2:11]: This file should be removed at the end of the role execution as it is not a file to expose to user.

**Review by titom73 — CHANGES_REQUESTED** [review]
Need to remove transient file at the end of the role: `generate_vars_for_testing.j2`

File: ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2 (hunk 1 above)
Comment (ksator) [reply ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2:11]: Done. This file is now removed at the end of the tests.

**Review by titom73 — APPROVED** [review]
Looks good to me!

**Pull request merged**